plugins/modules/cloud/misc/proxmox_nic.py

```diff
@@ -0,0 +1,334 @@
+        current_mac = config_current[model]
+
+        # build nic config string
+        config_provided = "{0}={1}".format(model, current_mac)
+    else:
+        config_provided = model
+
+    if kwargs['mac']:
+        config_provided = "{0}={1}".format(model, kwargs['mac'])
+
+    if kwargs['bridge']:
+        config_provided += ",bridge={0}".format(kwargs['bridge'])
+
+    if kwargs['firewall']:
+        config_provided += ",firewall=1"
+
+    if kwargs['link_down']:
+        config_provided += ',link_down=1'
+
+    if kwargs['mtu']:
+        config_provided += ",mtu={0}".format(kwargs['mtu'])
+
+    if kwargs['queues']:
+        config_provided += ",queues={0}".format(kwargs['queues'])
+
+    if kwargs['rate']:
+        config_provided += ",rate={0}".format(kwargs['rate'])
+
+    if kwargs['tag']:
+        config_provided += ",tag={0}".format(kwargs['tag'])
+
+    if kwargs['trunks']:
+        config_provided += ",trunks={0}".format(kwargs['trunks'])
+
+    net = {interface: config_provided}
+    vm = get_vm(proxmox, vmid)
+
+    if ((interface not in vminfo) or (vminfo[interface] != config_provided)):
+        if proxmox.nodes(vm[0]['node']).qemu(vmid).config.set(**net) is None:
+            return True
+
+    return False
+
+
+def delete_nic(proxmox, vmid, interface):
+    vm = get_vm(proxmox, vmid)
+    vminfo = proxmox.nodes(vm[0]['node']).qemu(vmid).config.get()
+
+    if interface in vminfo:
+        if proxmox.nodes(vm[0]['node']).qemu(vmid).config.set(vmid=vmid, delete=interface) is None:
+            return True
+
+    return False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_host=dict(required=True),
+            api_password=dict(no_log=True, fallback=(env_fallback, ['PROXMOX_PASSWORD'])),
+            api_token_id=dict(no_log=True),
+            api_token_secret=dict(no_log=True),
+            api_user=dict(required=True),
+            bridge=dict(type='str'),
+            firewall=dict(type='bool', default=False),
+            interface=dict(type='str', required=True),
+            link_down=dict(type='bool', default=False),
+            mac=dict(type='str'),
+            model=dict(choices=['e1000', 'e1000-82540em', 'e1000-82544gc', 'e1000-82545em',
+                                'i82551', 'i82557b', 'i82559er', 'ne2k_isa', 'ne2k_pci', 'pcnet',
+                                'rtl8139', 'virtio', 'vmxnet3'], default='virtio'),
+            mtu=dict(type='int'),
+            name=dict(type='str'),
+            queues=dict(type='int'),
+            rate=dict(type='float'),
+            state=dict(default='present', choices=['present', 'absent']),
+            tag=dict(type='int'),
+            trunks=dict(type='str'),
+            validate_certs=dict(type='bool', default=False),
+            vmid=dict(type='int'),
+        ),
+        required_together=[('api_token_id', 'api_token_secret')],
+        required_one_of=[('name', 'vmid'), ('api_password', 'api_token_id')],
```

Comment:
```suggestion
        required_one_of=[('name', 'vmid'), ('api_password', 'api_token_id')],
        supports_check_mode=True,
```
(And implement it.)

Reply:
I will look into this.

Reply:
check_mode implemented in the latest commit.